presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -136,6 +136,26 @@ public boolean add(Block block, int position)
         return false;
     }
 
+    public int addAndGetPosition(Block block, int position)
+    {
+        requireNonNull(block, "block must not be null");
+        checkArgument(position >= 0, "position must be >= 0");
+
+        // containsNullElement flag is maintained so contains() method can have shortcut for null value
+        if (block.isNull(position)) {
```

Comment:
this should handle containNullElements as well, right?

Reply:
done